apm-sdks: Document new set of configuration that should have consistent behavior across sdks (#28608)
DataDog/documentation: content/en/tracing/trace_collection/library_config/_index.md

`DD_ENV`
: **Default**: `null` <br>
**Supported Input**: A string representing an application environment name (for example, `prod`, `dev`) <br>
**Description**: Adds an environment tag to all spans generated by the tracer instance.

`DD_TAGS`
: **Default**: `null` <br>
**Supported Input**: A string representing key value pairs that are delimited by a colon and separated by a comma and/or a space (for example, `<key1>:<value1>, <key2>:<value2>`>) <br>
**Description**: Tags to apply to produced data. Must be a list of `<key>:<value>` separated by commas and/or spaces.

### Diagnostics

`DD_TRACE_LOG_DIRECTORY`
: **Default**: Varies by SDK, environment, and runtime. Please read more in the specific configuration page above for your chosen language <br>
**Supported Input**: A valid full or relative directory path that exists on the system<br>
**Caveats**: Not supported in Java, Node.js, Ruby, Python<br>
**Description**: Specifies the directory where tracer log files should be routed. If the directory does not exist, the SDK falls back to its default diagnostic logging method.

> 

Author: ```suggestion

### Metrics

```

> `DD_RUNTIME_METRICS_ENABLED`

Reviewer: nit: I don't think this belongs in the Diagnostics section. I think would be best classified under Metrics since this generates metrics that get sent to the Datadog platform

`DD_DOGSTATSD_PORT`
: **Default**: `8125` <br>
**Supported Input**: Integer between 0 and 65535 <br>
**Caveats**: Not supported in C++, Ruby, .NET<br>
**Description**: Specifies the port number for the DogStatsD server to which runtime metrics are sent.

`DD_DOGSTATSD_HOST`
: **Default**: `localhost` <br>
**Supported Input**: String representing a hostname or IP address <br>
**Caveats**: Not supported in C++, Ruby, .NET<br>
**Description**: Specifies the hostname or IP address of the DogStatsD server to which runtime metrics are sent.

`DD_TRACE_EXPERIMENTAL_FEATURES_ENABLED`
: **Default**: `null` <br>
**Supported Input**: A comma-separated list of configuration options that support experimental features (for example, `DD_TAGS,DD_RUNTIME_METRICS_ENABLED`) <br>
**Caveats**: Only supported in Java and .NET <br>
**Description**: Enables experimental features for specific configuration options. When enabled, these features may provide additional functionality but are not yet considered stable and may change or be removed in future releases. Each feature must be explicitly listed to be enabled.

> 

Reviewer 2: How are we planning to document which configs are supported under this `DD_TRACE_EXPERIMENTAL_FEATURES_ENABLED` config?

Author: I am not sure how to approach this but I think we have two options:
1) We can provide a high level description here and link to java/.net docs for more details
2) We list out all the supported values in this section of the doc

Reviewer 2: > We can provide a high level description here and link to java/.net docs for more details

This one makes more sense to me so we can handle variability of how each tracer might want to use this config. I left a comment below as well.

**Supported Input**: A string representing an application environment name (for example, `prod`, `dev`) <br>
**Description**: Adds an environment tag to all spans generated by the tracer instance.

> `DD_TAGS`

Reviewer: High-level comment: All of these changes are being captured in the `Unified Service Tagging` group of configurations. Can you reorganize them to fall in the appropriate sections? E.g. `DD_DOGSTATSD_PORT` and `DD_DOGSTATSD_HOST` should go into the Agent section of configurations

Author: I didn't even notice the section headers 😅 Updated

: **Default**: `null` <br>

> **Supported Input**: A string representing key value pairs that are delimited by a colon and seperate by a comma and/or a space (for example, `<key1>:<value1>, <key2>:<value2>`>) <br>

Author: ```suggestion
**Supported Input**:  A string of key-value pairs. Each pair has a key and value separated by a colon. Pairs are separated by commas or spaces. For example: <key1>:<value1>, <key2>:<value2>. <br>
```

**Description**: Tags to apply to produced data. Must be a list of `<key>:<value>` separated by commas and/or spaces.

`DD_TRACE_SAMPLING_RULES`
: **Default**: `null` <br>

> **Supported Input**: A JSON string representing an array of sampling rules. Each rule must contain a `sample_rate` (float between 0 and 1) and can optionally include a `service` (string) and `name` (string) pattern to match. Example: `[{"sample_rate": 0.1, "service": "my-service", "name": "http.request"}]` <br>

Reviewer: Our existing config pages have text that says `For more information, see [Ingestion Mechanisms][2]` with a link to `/tracing/trace_pipeline/ingestion_mechanisms/`. I think it would be helpful to include that here as well.

**Description**: Configures custom sampling rules for traces. Rules are evaluated in order, and the first matching rule determines the sampling rate. If no rules match, the default sampling rate is used.

`DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED`
: **Default**: `true` <br>
**Supported Input**: Boolean (`true`/`false`) <br>

> **Description**: Enables or disables the generation of 128-bit trace IDs. When enabled, trace IDs are generated as 32-character hexadecimal strings. When disabled, 64-bit trace IDs are used instead.

Reviewer: For both of the 128-bit toggles, I don't think it's immediately clear _why_ someone might want to set this. Can you add a description? Off the top of my head, I think something like `The latest Datadog tracing libraries support W3C trace IDs (128-bit) and generate new traces with W3C trace IDs by default. To generate trace IDs in the previous Datadog (64-bit) format, you can set this configuration to false.`

`DD_TRACE_128_BIT_TRACEID_LOGGING_ENABLED`
: **Default**: `true` <br>
**Supported Input**: Boolean (`true`/`false`) <br>
**Caveats**: Not supported in C++<br>
**Description**: Controls whether 128-bit trace IDs are logged in their full format. When disabled, only the lower 64 bits are logged.

`DD_TRACE_CLIENT_IP_ENABLED`
: **Default**: `true` <br>
**Supported Input**: Boolean (`true`/`false`) <br>
**Caveats**: Not supported in C++<br>
**Description**: Enables or disables the automatic collection of client IP addresses from HTTP request headers. When enabled, the IP address is stored in the `http.client_ip` span tag.

`DD_RUNTIME_METRICS_ENABLED`
: **Default**: `false` <br>
**Supported Input**: Boolean (`true`/`false`) <br>

> **Caveats**: Not supported in C++ and PHP<br>

Reviewer 3: ```suggestion
**Caveats**: Not supported in C++ or PHP<br>
```

> **Description**: Enables or disables the collection of runtime metrics (such as garbage collection stats, memory usage, and thread counts) for the application.

Reviewer: Can you link to the runtime metrics page in the description?

`DD_LOGS_INJECTION`
: **Default**: `false` <br>
**Supported Input**: Boolean (`true`/`false`) <br>

> **Caveats**: Not supported in C++ and Go. The default value in Ruby is `true`<br>

Reviewer 3: ```suggestion
**Caveats**: Not supported in C++ or Go. The default value in Ruby is `true`.<br>
```

**Description**: Enables or disables the automatic injection of trace context (trace ID, span ID) into application logs. This allows for correlation between traces and logs.

`DD_DOGSTATSD_PORT`
: **Default**: `8125` <br>
**Supported Input**: Integer between 0 and 65535 <br>
**Caveats**: Not supported in C++, Ruby, .NET<br>
**Description**: Specifies the port number for the DogStatsD server to which runtime metrics are sent.

`DD_DOGSTATSD_HOST`
: **Default**: `localhost` <br>
**Supported Input**: String representing a hostname or IP address <br>
**Caveats**: Not supported in C++, Ruby, .NET<br>
**Description**: Specifies the hostname or IP address of the DogStatsD server to which runtime metrics are sent.

`DD_TRACE_EXPERIMENTAL_FEATURES_ENABLED`
: **Default**: `null` <br>

> **Supported Input**: A comma-separated list of configuration options that support experimental features.

Reviewer 3: ```suggestion
**Supported Input**: A comma-separated list of configuration options that support experimental features.<br>
```

> **Supported Values**: `all`, `DD_TAGS` (java, .NET)

Reviewer 3: ```suggestion
**Supported Values**: `all`, `DD_TAGS` (java, .NET)<br>
```

> **Supported Values**: `all`, `DD_TAGS` (java, .NET), `DD_LOGS_INJECTION` (java) <br>

Reviewer 3: ```suggestion
**Supported Values**: `all`, `DD_TAGS` (Java, .NET), `DD_LOGS_INJECTION` (Java) <br>
```

: **Supported Input**: A regex string <br>
: **Description**: Applies a regex to redact sensitive data from query strings on incoming HTTP requests. The default regex matches various sensitive data patterns, including passwords, tokens, API keys, private keys, and authorization terms. Matches are replaced with `<redacted>`. If an empty string is passed, no obfuscation occurs. The resulting value is reported in the `http.url` tag.

`DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED`
: **Default**: `true` <br>
**Supported Input**: Boolean (`true`/`false`) <br>
**Description**: Controls whether new traces use 128-bit W3C trace IDs (32-character hexadecimal strings) or 64-bit Datadog trace IDs (16-character hexadecimal strings). The default is `true` to support W3C trace context propagation.

`DD_TRACE_128_BIT_TRACEID_LOGGING_ENABLED`
: **Default**: `true` <br>
**Supported Input**: Boolean (`true`/`false`) <br>

> **Caveats**: Not supported in C++<br>

Reviewer 4: Available in dd-trace-go v2 only.

Author: Do we plan to release v2 soon? If so, we can omit this detail. This page documents supported configurations just for the latest release

`DD_TRACE_EXPERIMENTAL_FEATURES_ENABLED`
: **Default**: `null` <br>
**Supported Input**: A comma-separated list of configuration options that support experimental features.

> **Supported Values**: `all`, `DD_TAGS` (java, .NET)

Reviewer 2: ```suggestion
**Supported Values**: `all`, `DD_TAGS` (java, .NET), `DD_LOGS_INJECTION` (java)
```

Albeit in the future I think this would make more sense to be documented on each languages' docs. Maybe this could also be made more clear through the Config Inversion OKR this quarter.

[1]: /developers/community/libraries/#apm-tracing-client-libraries
[2]: /tracing/trace_collection/compatibility/java/#framework-integrations-disabled-by-default
[3]: /tracing/services/inferred_services/
[4]: /tracing/trace_pipeline/ingestion_mechanisms/

> [5]: /tracing/runtime_metrics/

Reviewer: The previous `/tracing/runtime_metrics/` didn't work for me, but I think this should:

```suggestion
[5]: /tracing/metrics/runtime_metrics/
```
